### PR TITLE
Split native code into storage/engine/rocksdb

### PR DESCRIPTION
**Pull request by tamird**
This speeds up build times when a dependency of `engine` changes.
Before:

```
$ make build > /dev/null && touch storage/engine/engine.go && \
  time make build > /dev/null

real  0m20.892s
user  0m17.416s
sys 0m1.611s
```

After:

```
make build > /dev/null && touch storage/engine/engine.go && \
  time make build > /dev/null

real  0m6.348s
user  0m3.735s
sys 0m0.927s
```

### CODE REVIEW

File: storage/engine/rocksdb_native/rocksdb_native.go [hunk 1]
```diff
@@ -15,7 +15,26 @@
...
+// #cgo darwin LDFLAGS: -Wl,-undefined -Wl,dynamic_lookup
+// #cgo !darwin LDFLAGS: -Wl,-unresolved-symbols=ignore-all
+// #cgo linux LDFLAGS: -lrt
 import "C"
+
+//export rocksDBLog
+func rocksDBLog(s *C.char, n C.int) {
+	// Note that rocksdb logging is only enabled if log.V(1) is true
+	// when RocksDB.Open() is called.
+	log.Infof("%s", C.GoStringN(s, n))
```
Comment (petermattis) [inline storage/engine/rocksdb_native/rocksdb_native.go]: The dependency on `util/log` is unfortunate. You'll still rebuild `rocksdb_native` if `util/log` or any of its dependencies change. Might be worth adding something like:

```
var Logger = func(string, ...interface{}) {}
```

Then `storage/engine` can set `rocksdb_native.Logger = log.Infof`.

**Comment by petermattis** [comment]
Go style frowns on underscores in package names. Can we name this package `storage/engine/rocksdb`? Or is that too confusing with `cockroachdb/c-rocksdb`?

**Comment by petermattis** [comment]
LGTM

File: storage/engine/rocksdb_native/rocksdb_native.go (hunk 1 above)
Comment (tamird) [inline storage/engine/rocksdb_native/rocksdb_native.go]: Great idea, done.

**Comment by tamird** [comment]
> Can we name this package `storage/engine/rocksdb`?

Done.

File: build/protobuf.mk [hunk 2]
```diff
@@ -23,7 +23,7 @@ REPO_ROOT      := $(ORG_ROOT)/cockroach
 GITHUB_ROOT    := $(ORG_ROOT)/..
 GOGOPROTO_ROOT := $(GITHUB_ROOT)/gogo/protobuf
 
-ENGINE_ROOT := $(REPO_ROOT)/storage/engine
+NATIVE_ROOT := $(REPO_ROOT)/storage/engine/rocksdb
```
Comment (tschottdorf) [inline build/protobuf.mk]: should this be ROCKSDB_ROOT?

**Pull request merged**